syntax = "proto3";
package msg;

//protoc --go_out=. xxx.proto

// id : 0
message RegisterReq{
    string name = 1;
    string pass = 2;
}

// id : 1
message RegisterRsp{
    bool success = 1;
}


// id : 2
message LoginReq {
    string name = 1;
    string pass = 2;
}

// id : 3
message LoginRsp {
    bool success = 1;
    string name = 2;
    int32 rank = 3;
}

// id : 4, 请求匹配
message MatchReq {
    int32 type = 1; // 0 : 1v1, 1 : 2v2
}

message playerInfoBegin {
    int32 playerId = 1;
    string name = 2;
    int32 rank = 3;
}

// id : 5, 匹配成功
message MatchRsp {
    int32 playerNum = 1; //总玩家数
    int32 myPlayerId = 2; //该玩家编号 0 ~ playerNum - 1
    int32 roomId = 3;
    repeated playerInfoBegin players = 4;
}

// id : 6
message MatchCancelReq {

}

// id : 7
message MatchCancelRsp {
    bool success = 1;
}


